dowhy/datasets.py

```diff
@@ -399,3 +402,117 @@ def xy_dataset(num_samples, effect=True,
         "ate": None,
     }
     return ret_dict
+
+def create_discrete_column(num_samples,std_dev=1):
+    #Generating a random normal distribution of integers
+    x = np.arange(-5,6)
+    xU, xL = x + 0.5, x - 0.5 
+    prob = ss.norm.cdf(xU, scale = std_dev) - ss.norm.cdf(xL, scale = std_dev)  #probability of selecting a number x is p(x-0.5 < x < x+0.5) where x is a normal random variable with mean 0 and standard deviation std_dev
+    prob = prob / prob.sum() # normalize the probabilities so their sum is 1
+    nums = choice(a = x, size = num_samples, p = prob) #pick up an element 
+    return nums
+
+def convert_continuous_to_discrete(arr):
+    return arr.astype(int)
+
+
+def dataset_from_random_graph(num_vars, num_samples=1000, prob_edge=0.3, random_seed=100):
+    """
+    This function generates a dataset with discrete and continuous kinds of variables.
+    It creates a random graph and models the variables linearly according to the relations in the graph.
+
+    :param num_vars: Number of variables in the dataset
+    :param num_samples: Number of samples in the dataset
+    :param prob_edge : Probability of an edge between two random nodes in a graph
+    :param random_seed: Seed for generating random graph
+    """    
+    np.random.seed(109) 
+    G = nx.fast_gnp_random_graph(n = num_vars, p = prob_edge, seed= random_seed, directed=True) #Generating a random graph 
+    DAG = nx.DiGraph([(u,v) for (u,v) in G.edges() if u<v]) #Condition to maintain acyclicity
+    mapping = dict(zip(DAG, string.ascii_lowercase))
+    DAG = nx.relabel_nodes(DAG, mapping)
+    all_nodes = list(DAG.nodes)
+    all_nodes.sort()
+    num_nodes = len(all_nodes)
+    changed = dict()
+    discrete_cols = []
+    continuous_cols = []
+    random_numbers_array = np.random.rand(num_nodes) #Random numbers between 0 to 1 to decide if that particular node will be discrete or continuous
+
+    for node in all_nodes:
+        changed[node] = False
+    df = pd.DataFrame()
+    currset = list()
+    counter = 0
+
+    #Generating data for nodes which have no incoming edges
+    for node in all_nodes:
+        if DAG.in_degree(node) == 0:
+            x = random_numbers_array[counter]  
+            counter+=1
+            if x<=0.333:
```

Comment:
can you make this 0.333 as an argument for the function? It can be a tuple that provides the probability of variable being continuous, binary, discrete. E.g. (0.33, 0.33, 0.33) or (0.5. 0.4. 0.1). We ask the user to ensure that they sum to 1.

Reply:
(default can be 0.33)